lib/ansible/modules/packaging/os/yum.py

```diff
@@ -1029,6 +1029,10 @@ def remove(module, items, repoq, yum_basecmd, conf_file, en_repos, dis_repos, en
                                          dis_plugins=dis_plugins, installroot=installroot)
 
             if installed:
+                # Return a special 'error' field so it's obvious to the user
+                # why yum failed and which package couldn't be removed. More
+                # details: https://github.com/ansible/ansible/issues/35672
+                res['error'] = "Package '%s' couldn't be removed!" % pkg
```

Comment:
I thought about it, but in this situation the `msg` looks like this:
```
Skipping the running kernel: kernel-3.10.0-693.21.1.el7.x86_64
```
Which isn't that obvious tbh and if I appended my message it would be easy to miss it. I would leave `msg` to the `yum` output.

Also `yum` module is all over the place regarding the return keys, it would really be nice if we made it consistent and also add the docs about the return keys. I could probably provide some proposal before any work is done on it if you are interested.

Reply:
I don't have strong feelings about this really but I guess my concern was that with this we could end up with having error messages in both `msg` and `error` at the same time which seems little confusing to me.

Reply:
Having said that we could do this (add `error`) for now and do cleanup for the whole module (including documenting the return keys) later.

Reply:
if it is an error, use fail_json, otherwise it is a warning, which `module.warn()`  can take care of

Reply:
@bcoca I don't understand what you are saying. We are using `fail_json`, it's in the next line. This will just make sure all the fields which were set before are also returned.

Reply:
then that should be the `msg=` to fail_json, the `error` key is not a good choice here, i could see a `'failed_pkgs' = [ list of failed]` as an additional key, but `error` seems to be part of a standard that does not exist.

Reply:
Agreed with @bcoca here, I would like to see that land in `msg`